### PR TITLE
tinymist: init at 0.11.1

### PR DESCRIPTION
**Pull request by LamprosPitsillos**
## Description of changes
init `tinymist`
solves #296943

Tinymist [ˈtaɪni mɪst] is an integrated language service for Typst [taɪpst]. You can also call it "微霭" [wēi ǎi] in Chinese.

It contains:

an analyzing library for Typst, see tinymist-query.
a CLI for Typst, see tinymist. which provides a language server for Typst.
a VSCode extension for Typst, see Tinymist VSCode Extension.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — CHANGES_REQUESTED** [review]
Please fix the darwin builds.

Tips:
1. Look at the error message and search `nixpkgs` to find similar packages (there are dozens)
2. If you don't have a Darwin machine, just push your changes, the CI will build on Darwin for you
3. If you really can't find a solution, you can ask @NixOS/darwin-maintainers team

File: pkgs/by-name/ti/tinymist/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+         "typst-0.11.0" = "sha256-UzZ0tbC6Dhn178GQDyLl70WTp3h5WdaBCsEKgLisZ2M=";
+    };
+  };
+
+  meta = with lib; {
+    description = "";
+    homepage = "https://github.com/Myriad-Dreamin/tinymist ";
```
Comment (Myriad-Dreamin) [inline pkgs/by-name/ti/tinymist/package.nix:26]: trailing space here?

File: pkgs/by-name/ti/tinymist/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    hash = "sha256-UteAlLJFIgLBcHfQqcBcNR84JOMG9wxHyi8FyXLktWg=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+         "typst-0.11.0" = "sha256-UzZ0tbC6Dhn178GQDyLl70WTp3h5WdaBCsEKgLisZ2M=";
+    };
+  };
+
```
Comment (viraptor) [inline pkgs/by-name/ti/tinymist/package.nix:23]:
```suggestion
  buildInputs = with darwin.apple_sdk_11_0.frameworks; lib.optionals (stdenv.isDarwin) [
    CoreServices
    Security
    SystemConfiguration
  ];
```

File: pkgs/by-name/ti/tinymist/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
```
Comment (viraptor) [inline pkgs/by-name/ti/tinymist/package.nix:3]:
```suggestion
, fetchFromGitHub
, darwin
```

File: pkgs/by-name/ti/tinymist/package.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ lib
```
Comment (viraptor) [inline pkgs/by-name/ti/tinymist/package.nix:1]:
```suggestion
{ lib
, stdenv
```

**Review by viraptor — COMMENTED** [review]
That's enough to get darwin running

**Review by drupol — CHANGES_REQUESTED** [review]
The commit related to adding you as maintainer should come first, not last.

**Review by drupol — APPROVED** [review]

File: pkgs/by-name/ti/tinymist/package.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  ];
+
+  buildInputs = [
+    libgit2
+    openssl
+    zlib
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.CoreFoundation
+    darwin.apple_sdk.frameworks.CoreServices
+    darwin.apple_sdk.frameworks.Security
```
Comment (viraptor) [inline pkgs/by-name/ti/tinymist/package.nix:41]: It has to be the newer version of the SDK or darwin x86 will fail.
```suggestion
    darwin.apple_sdk_11_0.frameworks.CoreFoundation
    darwin.apple_sdk_11_0.frameworks.CoreServices
    darwin.apple_sdk_11_0.frameworks.Security
```

File: pkgs/by-name/ti/tinymist/package.nix [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  ];
+
+  buildInputs = [
+    libgit2
+    openssl
+    zlib
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk_11_0.frameworks.CoreFoundation
+    darwin.apple_sdk_11_0.frameworks.CoreServices
+    darwin.apple_sdk_11_0.frameworks.Security
```
Comment (viraptor) [inline pkgs/by-name/ti/tinymist/package.nix:41]: I don't know why this wasn't required to build on my system, but still seems to be needed for x86 build according to ofborg.
```suggestion
    darwin.apple_sdk_11_0.frameworks.Security
    darwin.apple_sdk_11_0.frameworks.SystemConfiguration
```

Comment (LamprosPitsillos) [reply pkgs/by-name/ti/tinymist/package.nix:41]: Thanks for all the suggestions! Hope all works now

**Review by wegank — APPROVED** [review]

File: pkgs/by-name/ti/tinymist/package.nix (hunk 5 above)
Comment (drupol) [reply pkgs/by-name/ti/tinymist/package.nix]: It builds fine on the CI, is your comment still relevant?

**Comment by wegank** [comment]
I believe the comment has been addressed in https://github.com/NixOS/nixpkgs/commit/37f49a2579e130ddd837cf22428c9cab731d0367.

**Pull request merged**